### PR TITLE
Allow for agent configs to be downloaded from http(s) urls

### PR DESCRIPTION
**Pull request by steiler**
implementing #1331

### CODE REVIEW

**Comment by hellt** [comment]
Hi @steiler I am ok with that change, can you rebase it on current `main` so that srl pipeline gets to working status again?

**Comment by steiler** [comment]
ok, it is rebased.
I've also added some code that will add the .yml extension if no filename extension is provided but also change it to .yml if it is not already .yaml or .yml

File: nodes/srl/srl.go [hunk 1]
```diff
@@ -209,11 +209,20 @@ func (s *srl) PreDeploy(_ context.Context, params *nodes.PreDeployParams) error
...
 		for _, fullpath := range agents {
 			basename := filepath.Base(fullpath)
+			// if it is a url extract filename from url or content-disposition header
+			if utils.IsHttpUri(fullpath) {
+				basename = utils.FilenameForURL(fullpath)
+			}
+			// enforce yml extension
+			if ext := filepath.Ext(basename); !(ext == ".yml" || ext == ".yaml") {
+				basename = basename + ".yml"
+			}
```
Comment (hellt) [inline nodes/srl/srl.go:224]: @steiler this is likely a simpler way?

Comment (steiler) [reply nodes/srl/srl.go:224]: Rome and all the streets that lead there and so ... ;-)

**Pull request merged**